core/desugarer.cpp

```diff
@@ -310,14 +317,21 @@ class Desugarer {
                     expr = alloc.make<Var>(expr->location, expr->openFodder, newSelf);
                 } else if (auto *super_index = dynamic_cast<SuperIndex*>(expr)) {
                     StringStream ss;
-                    ss << "$outer_super" << (counter++);
+                    ss << "$outer_super_index" << (counter++);
```

Comment:
Oh looks like I lied:  Identifiers are UTF32 which is why it uses StringStream in this case.  However the literals should also be UTF32 and I'm not sure why the compiler didn't complain here.  I guess because << is more permissive than std::base_string's + operator.  I will fix the literals.